modules/class.jetpack-calypsoify.php

```diff
@@ -0,0 +1,195 @@
+				'closeUrl'   => $this->get_close_gutenberg_url(),
+			)
+		);
+	}
+
+	public function insert_sidebar_html() { ?>
+		<a href="<?php echo esc_url( 'https://wordpress.com/stats/day/' . str_replace( 'https://', '', get_bloginfo( 'url' ) ) ); ?>" id="calypso-sidebar-header">
+		  <svg class="gridicon gridicons-chevron-left" height="24" width="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><g><path d="M14 20l-8-8 8-8 1.414 1.414L8.828 12l6.586 6.586"></path></g></svg>
+
+		  <ul>
+			<li id="calypso-sitename"><?php bloginfo( 'name' ); ?></li>
+			<li id="calypso-plugins"><?php esc_html_e( 'Plugins' ); ?></li>
+		  </ul>
+		</a>
+		<?php
+	}
+
+	public function modify_masterbar() {
+		global $wp_admin_bar;
+
+		// Add proper links to masterbar top sections.
+		$my_sites_node       = $wp_admin_bar->get_node( 'blog' );
+		$my_sites_node->href = 'https://wordpress.com/stats/day/' . str_replace( 'https://', '', get_bloginfo( 'url' ) );
+		$wp_admin_bar->add_node( $my_sites_node );
+
+		$reader_node       = $wp_admin_bar->get_node( 'newdash' );
+		$reader_node->href = 'https://wordpress.com';
+		$wp_admin_bar->add_node( $reader_node );
+
+		$me_node       = $wp_admin_bar->get_node( 'my-account' );
+		$me_node->href = 'https://wordpress.com/me';
+		$wp_admin_bar->add_node( $me_node );
+	}
+
+	private function installed_plugins_icon() {
+		$svg = '<svg class="gridicon gridicons-plugins" height="24" width="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 24"><g><path d="M16 8V3c0-.552-.448-1-1-1s-1 .448-1 1v5h-4V3c0-.552-.448-1-1-1s-1 .448-1 1v5H5v4c0 2.79 1.637 5.193 4 6.317V22h6v-3.683c2.363-1.124 4-3.527 4-6.317V8h-3z" fill="black"></path></g></svg>';
+
+		return 'data:image/svg+xml;base64,' . base64_encode( $svg );
+	}
+
+	public function get_close_gutenberg_url() {
+		$screen = get_current_screen();
+
+		// E.g. `posts`, `pages`, or `types/some_custom_post_type`
+		$post_type = ( 'post' === $screen->post_type || 'page' === $screen->post_type )
+			? $screen->post_type . 's'
+			: 'types/' . $screen->post_type;
+
+		return 'https://wordpress.com/' . $post_type . '/' . Jetpack::build_raw_urls( home_url() );
+	}
+
+	public function check_param() {
+		if ( isset( $_GET['calypsoify'] ) ) {
+			if ( 1 == (int) $_GET['calypsoify'] ) {
+				update_user_meta( get_current_user_id(), 'calypsoify', 1 );
+			} else {
+				update_user_meta( get_current_user_id(), 'calypsoify', 0 );
+			}
+
+			$page = remove_query_arg( 'calypsoify', wp_basename( $_SERVER['REQUEST_URI'] ) );
+
+			wp_safe_redirect( admin_url( $page ) );
+		}
+	}
+
+	public function check_page() {
+		// If the user hits plain /wp-admin/ then disable Calypso styles.
+		$page = wp_basename( esc_url( $_SERVER['REQUEST_URI'] ) );
+
+		if ( false !== strpos( 'index.php', $page ) || false !== strpos( 'wp-admin', $page ) ) {
+			update_user_meta( get_current_user_id(), 'calypsoify', 0 );
+			wp_safe_redirect( admin_url() );
+			die;
+		}
+	}
+
+	public function is_page_gutenberg() {
```

Comment:
This needs some post type logic. Otherwise, the sidebar completely disappears on pages where Gutenberg isn't actually used (like the WooCommerce New Orders page).